.github/workflows/codecov.yml

```diff
@@ -0,0 +1,108 @@
+name: Code Coverage
+on: [push, pull_request]
+concurrency:
+  group: format('{0}-{1}', ${{ github.ref }}, 'Code Coverage')
+  cancel-in-progress: true
+
+permissions: read-all
+
+jobs:
+  test:
+    name: Code Coverage
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Check for changes in files relevant to code coverage
+      uses: frouioui/paths-filter@main
+      id: changes
+      with:
+        token: ''
+        filters: |
+          go_files:
+            - 'go/**'
```

Comment:
We should add more to the list of files that trigger this workflow:
- this workflow file `.github/workflows/codecov.yml` 
- `./Makefile` as it is being used by this workflow to run the tests
- `go.mod` and `go.sum`

Reply:
Makes sense. Thanks!